docs/source/howto/plugin_codes.rst

```diff
@@ -223,104 +250,205 @@ The Topics section on :ref:`defining processes <topics:processes:usage:defining>
-You can use the verdi command line interface to :ref:`monitor<topics:processes:usage:monitoring>` these processes:
 
-.. code-block:: bash
+So far, we have deliberately avoided to touch the concept of :ref:`entry points <how-to:plugins:entrypoints>`, which are the preferred method of announcing new calculations, parsers and other plugins to AiiDA.
+Using entry points simplifies your life in a number of ways:
 
-    $ verdi process list
+ * You can use ``verdi plugin list`` to inspect which plugins are available as well as their internal documentation.
+ * You can specify a default |Parser| for a |CalcJob|, which will run automatically.
+ * You can specify a default |CalcJob| for a |Code|, making it easy to set the inputs for a code.
+ * You can submit calcualtions to the AiiDA daemon without having to worry about how to make sure that your calculation and parser plugins can be imported from the daemon's python environment.
 
-Performing a dry-run
---------------------
+Registering entry points
+------------------------
 
-Additionally, you might want to check and verify your inputs before actually running or submitting a calculation.
-You can do so by specifying to use a ``dry_run``, which will create all the input files in a local directory (``submit_test/[date]-0000[x]``) so you can inspect them before actually launching the calculation:
+Registering entry points is simple:
 
-.. code-block:: python
+ * Write a minimalistic ``setup.py`` script:
 
-    calculation_builder.metadata.dry_run = True
-    calculation_builder.metadata.store_provenance = False
-    run(calculation_builder)
+    .. code-block:: python
 
-.. todo::
+        from setuptools import setup
+
+        setup(
+            name='aiida-add',
+            packages=['aiida_add'],
+            entry_points={
+                'aiida.calculations': ["add = aiida_add.calcjob:ArithmeticAddCalculation"],
+                'aiida.parsers': ["add = aiida_add.parser:ArithmeticAddParser"],
+            }
+        )
+
+ * Move your two scripts into a subfolder ``aiida_add``:
+
+   .. code-block:: console
+
+      mkdir aiida_add
+      mv calcjob.py parser.py aiida_add/
+
+ * Install your new ``aiida-add`` plugin package:
+
+   .. code-block:: console
+
+      pip install -e .
+      reentry scan
+
+
+After this, you should already see your calculation plugin listed:
+
+   .. code-block:: console
+
+      $ verdi plugin list aiida.calculations
+      $ verdi plugin list aiida.calculations add
+
+
+Launching your calculation now becomes easier:
+
+ * In ``calcjob.py``, change the default  ``parser_name`` to ``'add'``
+
+ * Simplify your ``launch.py`` script:
+
+    .. code-block:: python
+
+        from aiida import orm, engine
 
-    .. _how-to:plugin-codes:caching:
+        # Setting up inputs
+        computer = orm.load_computer('localhost')
+        code = orm.Code(remote_computer_exec=[computer, '/bin/bash'], input_plugin_name='add')
```

Comment:
This is actually adding complexity, adding the `input_plugin_name` right? And there is no benefit to it here.

Reply:
> And there is no benefit to it here.

This is not true. Adding the input plugin is what allows using the builder.